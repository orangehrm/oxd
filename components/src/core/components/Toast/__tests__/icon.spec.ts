import {mount} from '@vue/test-utils';
import ToastIcon from '@orangehrm/oxd/core/components/Toast/Icon.vue';

describe('Toast > Icon.vue', () => {
  it('should renders OXD ToastIcon', () => {
    const wrapper = mount(ToastIcon, {
      props: {},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD ToastIcon success', () => {
    const wrapper = mount(ToastIcon, {
      props: {type: 'success'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD ToastIcon warn', () => {
    const wrapper = mount(ToastIcon, {
      props: {type: 'warn'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD ToastIcon error', () => {
    const wrapper = mount(ToastIcon, {
      props: {type: 'error'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD ToastIcon info', () => {
    const wrapper = mount(ToastIcon, {
      props: {type: 'info'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
