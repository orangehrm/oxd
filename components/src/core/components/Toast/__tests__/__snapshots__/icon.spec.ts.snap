// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast > Icon.vue should renders OXD ToastIcon 1`] = `
<div class="oxd-toast-icon-container">
  <div class="oxd-toast-icon-wrap oxd-toast-icon-wrap--default"><i class="oxd-icon oxd-icon--medium bi-star oxd-toast-icon"></i></div>
</div>
`;

exports[`Toast > Icon.vue should renders OXD ToastIcon error 1`] = `
<div class="oxd-toast-icon-container">
  <div class="oxd-toast-icon-wrap oxd-toast-icon-wrap--error"><i class="oxd-icon oxd-icon--medium bi-exclamation-circle oxd-toast-icon"></i></div>
</div>
`;

exports[`Toast > Icon.vue should renders OXD ToastIcon info 1`] = `
<div class="oxd-toast-icon-container">
  <div class="oxd-toast-icon-wrap oxd-toast-icon-wrap--info"><i class="oxd-icon oxd-icon--medium bi-info-circle oxd-toast-icon"></i></div>
</div>
`;

exports[`Toast > Icon.vue should renders OXD ToastIcon success 1`] = `
<div class="oxd-toast-icon-container">
  <div class="oxd-toast-icon-wrap oxd-toast-icon-wrap--success"><i class="oxd-icon oxd-icon--medium bi-check2 oxd-toast-icon"></i></div>
</div>
`;

exports[`Toast > Icon.vue should renders OXD ToastIcon warn 1`] = `
<div class="oxd-toast-icon-container">
  <div class="oxd-toast-icon-wrap oxd-toast-icon-wrap--warn"><i class="oxd-icon oxd-icon--medium bi-exclamation-triangle oxd-toast-icon"></i></div>
</div>
`;
